### DISCUSSION TITLE
Env variable is completely ignored in template checks

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3387] [Q&A] [answered]
I'm trying to display an announcement banner depending on if the docs were build with GitHub Actions or not.

For that did I add the following option to the `mkdocs.yml` file:
```yaml
extra:
  github_repository: !ENV [GITHUB_REPO_INSTANCE, false]
```

In the `main.html` did I add this:
```jinja2
{% if config.extra.github_repository %}
  {% block announce %}
    This documentation has been moved!
  {% endblock %}
{% endif %}
```
Now, no matter if the option is true, or false is the banner displayed, despite the if-check.

Just to check did I print out `{{ config.extra.github_repository }}` and the output was `False`, so I changed the if-check to `{% if config.extra.github_repository and (config.extra.github_repository == True) %}` with no change whatsoever...
Changing it to check if output is not `False` also doesn't work...

Why does this happen? What am I doing wrong here?

**Comment by kamilkrzyskow** [comment, marked as answer]
I've seen almost the same issue a while ago either here or in the Material theme discussion, as Material defines the announce block to override.

The issue is likely the order of your Jinja2 tags. 

Jinja processes the block extension first, so as you have no if conditional inside the block extension it always works no matter what is wrapped around the block extension.

So put the conditional inside the block extension.

**Reply by Andre601** [reply]
That *kinda* works... Issue is, that if it is false, then there is a small black bar, since the announce block is there, but no content is in it... :/

Guess that system isn't made for conditional setups like that.

**Reply by Andre601** [reply]
I got a workaround by overriding the content partial and insert a admonition box at the top to inform about the move.

**Reply by kamilkrzyskow** [reply]
The announce block is always there:

https://github.com/squidfunk/mkdocs-material/blob/502a517e2e7774c0518a60f0c8bf502b25671284/material/base.html#L123

And I do use it with a conditional too:

https://github.com/Gothic-Modding-Community/gmc/blob/8e15b7922228dc961557c9f1570eba14ffb47e32/overrides/main.html#L35

and I don't remember having any issues 🤔, apart of the case where I added a `<div>` without any text inside, but that's not the case in your example because you only add text without any `<div>`, hmm. 
However, note that for the time being I don't use the newest version of the Material theme (still on 9.1.21), though I doubt it matters.

You can always check the source of the rendered HTML to see what's going on that adds that small black bar. 
Or check the layout of the announce section in the browser Developer Tools to see what adds the black bar.

**Reply by Andre601** [reply]
Perhaps it's the fact that no div or other html element is present so it renders an empty p tag or smth...

**Reply by kamilkrzyskow** [reply]
> I got a workaround by overriding the content partial and insert a admonition box at the top to inform about the move.

I'm not really a fan of overriding whole html files, as it adds things to look out for, when updating the theme etc.
When you really have to modify html files outside of the overrideable blocks, I would rather do it dynamically like so:
https://github.com/squidfunk/mkdocs-material/discussions/5060
Though, please note I'm close to releasing a refactored approach to what was written there.

**Reply by kamilkrzyskow** [reply]
> Perhaps it's the fact that no div or other html element is present so it renders an empty p tag or smth...

If no text gets added, because the conditional isn't True, then it shouldn't add any `<p>` tag either. Just some empty white space.

**Reply by Andre601** [reply]
> I'm not really a fan of overriding whole html files, as it adds things to look out for, when updating the theme etc. When you really have to modify html files outside of the overrideable blocks, I would rather do it dynamically like so: [squidfunk/mkdocs-material#5060](https://github.com/squidfunk/mkdocs-material/discussions/5060) Though, please note I'm close to releasing a refactored approach to what was written there.

It's not that big of a deal for me.
I already have to override some files here. Like the source file to display stats from Codeberg which requires some Javascript to be added alongside using a proxy to fetch data through the API.
